repositories/EventRepository.ts

```diff
@@ -40,9 +40,21 @@ export class EventRepository extends BaseRepository<EventModel> {
     return this.repository.remove(event);
   }
 
-  public async isUnusedAttendanceCode(attendanceCode: string): Promise<boolean> {
-    const count = await this.repository.count({ attendanceCode });
-    return count === 0;
+  public async isAvailableAttendanceCode(attendanceCode: string, start: Date, end: Date): Promise<boolean> {
+    const duringCurrentEvent = await this.repository.find({
+      where: [
```

Comment:
Hold up looking back, this filters all the events that starts or ends between the new event period. Does this account for preexisting events that overlap with the entire new event period?

I think a better logic checking conflicting events would be: events that start before the new event's end time AND end after the new event's start time.

Reply:
I originally had it like that, but there might be some edge cases. 

For example, if there's an event 2PM - 3PM, that logic wouldn't mark an event 1PM - 2:30PM as conflicting, and would allow both to be created. Then, using the check-in code at 1:15PM for instance would result in unexpected behavior.

Reply:
Jimmy's thing queries for start and end each being between the new start and end times, so wouldn't it work? If a NEW event is conflicting it would either start in the middle of an old event or end before a new event ended, which is what we're checking right? 

Unrelated edge case but idk if Between is inclusive so if someone accidentally creates a perfectly conflicting event not sure if it'd be caught